src/CombinePDF/CombinePDF.tsx

```diff
@@ -80,6 +97,35 @@ const CombinePDF = (): JSX.Element => {
     );
   }, [doc, fileName, pagesOrder]);
 
+  const onDelete = async (pageNumber: number) => {
+    if (!doc) return;
+
+    const currentPages = new Array(doc.getPageCount())
+      .fill(0)
+      .map((_, index) => index)
+      .filter((page) => page !== pageNumber - 1);
+
+    // A document needs to at least have one page
+    // @(TODO) show error message
+    if (currentPages.length < 1) return;
```

Comment:
I mentioned below, we can either hide the delete button when there's only one page, OR let people delete all pages and do something line 

```suggestion
    if (currentPages.length < 1) {
      setPdf(undefined);
      setDoc(undefined);
      setPagesOrder([]);
      return;
    }
```

That should automatically create a new document next time they upload a new file.